lib/Test/Nginx/Util.pm

```diff
@@ -2379,6 +2389,36 @@ END {
     }
 }
 
+sub map_test_name_to_path($$) {
+    my ($archive_path, $name) = @_;
+    # convert t/archive.t "TEST 1: hello world" to t.archive.TEST_1:_hello_world
+    (my $file = $0) =~ s/\.[^.]+$/\./;
+    $file =~ s/\//\./g;
+    $name =~ s/\s/_/g;
+    return File::Spec->catfile($archive_path, $file.$name);
+}
+
+sub archive_files($$$) {
+    my ($name, $src) = @_;
+    my $dest = map_test_name_to_path($ArchivePath, $name);
```

Comment:
I think you need to take into account the case that the destination directory already exists. It actually can happen when 2 test blocks in the same file have the same title.